common/extensions.go

```diff
@@ -30,10 +31,18 @@ type URLExtension struct {
 // URLWithPlusDecodedInPath returns a URL with '+' in path decoded as ' '(space).
 // This is useful for the cases, e.g: S3 management console encode ' '(space) as '+', which is not supported by Azure resources.
 func (u URLExtension) URLWithPlusDecodedInPath() url.URL {
-	if u.RawPath != "" && strings.Contains(u.RawPath, "+") {
-		u.RawPath = strings.Replace(u.RawPath, "+", " ", -1)
-		u.Path = url.PathEscape(u.RawPath)
+	// url.RawPath is not always present. Which is likely, if we're _just_ using +.
+	if u.RawPath != "" {
+		var err error
+		u.RawPath = strings.ReplaceAll(u.RawPath, "+", "%20")
+		u.Path, err = url.PathUnescape(u.RawPath)
+
+		PanicIfErr(err)
```

Comment:
In theory, this may not be _totally_ safe, because it makes an assumption about RawPath that the core Go library tries to avoid. Namely it assumes that, on entry to this function, RawPath is either null or is a valid encoding of Path.  THat's the way things are supposed to be, be the Go SDK carefully avoids making that assumption.  Insetead, the Go SDK help says this:

>In general, code should call EscapedPath instead of reading u.RawPath directly.

EscapedPath returns RawPath if it exists, and if it is valid.  Else it returns Path.

So, technically, I _think_ we should make sure RawPath is valid, by changing the if statement here to this:
```
if u.RawPath != "" && u.RawPath == u.EscapedPath() {   // only look at RawPath if it exists AND is valid
```

I'm not sure how necessarry it really is to do this.  I feel nervous when we make an assumption that the Go SDK deliberately doesn't.  What do you think?

Reply:
I would say that if the rawpath were invalid and we were this far, that's something we should panic over.

Why? Because we've already lost data about what the user wanted at that point. We can't verify any individual character of RawPath means anything with the context of Path.

Reply:
Furthermore, it'd act as a safeguard for any more URL manipulation we end up doing prior to this point in the future

Reply:
OK. So this would mean inside the existing if, doing this:

```
if u.RawPath != u.EscapedPath() {
   panic(…)
}
```

It feels mildly unconventional, but I'm comfortable with you doing it.